src/Microsoft.SqlTools.ServiceLayer/FileBrowser/FileBrowserService.cs

```diff
@@ -158,14 +159,26 @@ public void InitializeService(ServiceHost serviceHost)
             RequestContext<FileBrowserCloseResponse> requestContext)
         {
             FileBrowserCloseResponse response = new FileBrowserCloseResponse();
-            if (this.ownerToFileBrowserMap.ContainsKey(fileBrowserParams.OwnerUri))
+            try
             {
-                this.ownerToFileBrowserMap.Remove(fileBrowserParams.OwnerUri);
-                response.Succeeded = true;
+                if (this.ownerToFileBrowserMap.ContainsKey(fileBrowserParams.OwnerUri))
+                {
+                    FileBrowserOperation browserOperation = this.ownerToFileBrowserMap[fileBrowserParams.OwnerUri];
+                    if (!browserOperation.FileTreeCreated)
+                    {
+                        browserOperation.Cancel();
+                    }
+                    else
+                    {
+                        browserOperation.SqlConnection.Close();
+                    }
+                    this.ownerToFileBrowserMap.Remove(fileBrowserParams.OwnerUri);
```

Comment:
shouldn't you use concurrent dictionary? what happens if two request to close comes at the same time?